### ISSUE TITLE
How much can I mess here?

### ISSUE THREAD

**Issue by skalee** [metanorma/reverse_adoc#55]
I'm going to use this gem in https://github.com/glossarist/iev-data, but I need to improve it first. Before that I want to ask how much can I mess here.

For example:

1. Rubocop reports plenty of code style issues and I'm going to either fix them or to update the style book for this project to match actual style. I guess the former is a better option, however it will result with merge conflicts if someone has some work-in-progress branches.
2. I'll likely do some refactoring, and this will add another merge conflicts if someone has WIP branches.
3. This gem has many dependencies, some of them are needed only for Word documents. I am thinking of making them optional, or maybe even extracting Word handling to a separate gem. This means a breaking change for users.
4. I may change some defaults after adding new features. This means more breaking changes.
5. And more.

The bottom line is that I'm going to make significant changes here, I believe that beneficial ones. However, some of them may interrupt others' work, some other may be controversial. Can I proceed with such changes or should I refrain from them? Should I consult functional changes in advance? Should I ask for pull request reviews?

/cc @ronaldtse @opoudjis @CAMOBAP @w00lf @strogonoff

**Comment by ronaldtse** [comment]
Please proceed with the changes, this gem is due for refactoring. 

The automated workflow that depends on this gem is the annotated-express work, so you could tag @w00lf and @opoudjis for reviews since they are most involved.

**Comment by w00lf** [comment]
Hi there, currently we dont have any active work here so you can proceed safely with the refactoring.

**Comment by opoudjis** [comment]
This gem is... not abandonware, exactly, but hasn't been touched in a while. @w00lf, several of your updates to the stepmod script should be ported back here, as I highlighted at the time, but it is not pressing.

I will object virulently to any intervention in the metanorma-* gems and their core associates (isodoc, html2doc), though I did just install ALE in vim to expedite future rubocop cleanup.

**Comment by ronaldtse** [comment]
@opoudjis it's not abandonware -- our editors are using this, as well as the Express work. @w00lf has already ported back a lot of functionality here.

**Comment by skalee** [comment]
@opoudjis Instead of doing clean-ups, I can simply disable some Rubocop checks so that it will be far less verbose. I have no problem with that.

**Comment by skalee** [comment]
I've just disabled several Rubocop checks for this project (#62). However, following questions remain valid, I guess they all are addressed to @opoudjis:

- Should I refactor or should I refrain from doing that?
- Should I consult functional changes in advance?
- Should I ask for pull request reviews?

> I will object virulently to any intervention in the metanorma-* gems and their core associates (isodoc, html2doc).

Then I understand that answer for all above is "yes".

**Comment by opoudjis** [comment]
I'm doing the Rubocop updates myself in the metanorma stack, slowly, now that I've incorporated Rubocop into my text editor. So the answer is yes, but I'm doing it. But for this gem, do whatever you want.

**Comment by ronaldtse** [comment]
Closing.
